controllers/nodehealthcheck_controller_test.go

```diff
@@ -363,12 +371,23 @@ var _ = Describe("Node Health Check CR", func() {
 				err := k8sClient.Get(context.Background(), client.ObjectKeyFromObject(cr), cr)
 				Expect(errors.IsNotFound(err)).To(BeTrue())
 
-				// TODO Expect(reconcileResult.RequeueAfter).To(Equal(1 * time.Minute))
 				Expect(underTest.Status.HealthyNodes).To(Equal(2))
 				Expect(underTest.Status.ObservedNodes).To(Equal(3))
 				Expect(underTest.Status.InFlightRemediations).To(HaveLen(0))
 				Expect(underTest.Status.Phase).To(Equal(v1alpha1.PhaseEnabled))
 				Expect(underTest.Status.Reason).ToNot(BeEmpty())
+
+				By("stopping upgrade and waiting for requeue")
+				upgradeChecker.Upgrading = false
+				// the fake time won't work here, because the reconciler internals do noy use it
+				time.Sleep(65 * time.Second)
```

Comment:
IMO we should do an effort to avoid such a timeout in UT.
One way to do it is using a global var instead of internal field to get the time (which I think will be easier to be called by the  reconciler internals)

something like
```
var getTimeNow = func() time.Time {
	return time.Now()
}
```

Reply:
how do you make controller-runtime use that function for determing that the requeue time is over?

Reply:
(in general I agree that it's not nice, but the value of running this test is greater than the disadvantage of a one minute longer test time IMHO)

Reply:
The problem is that we can not mock the current time in the internals of controller-runtime, where it decides that the requeue time has passed for triggering a new reconcile. The problem is not in our own comtroller code.

Reply:
I think I understand, thanks :+1: 
So IIUC it is the exponential back-off timeout that we having an issue with and not a requeue timeout we decide on within our controller code, right ?

Reply:
No, not exponential, this is testing the fixed requeue of 1 minute during cluster upgrades
https://github.com/medik8s/node-healthcheck-operator/blob/main/controllers/nodehealthcheck_controller.go#L227

Reply:
In that case, maybe we can mock the requeue time for UT purpose ?

Reply:
done